setup/react/app/ApplicationBootstrap.js

```diff
@@ -1,46 +1,63 @@
 /* eslint-disable complexity */
 /* eslint-disable max-statements */
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { useTransactionUpdate } from '@transaction/hooks';
 import useSettings from '@settings/hooks/useSettings';
-import { useGetMergedApplication } from '@blockchainApplication/manage/hooks/useGetMergedApplication';
-import { useCurrentApplication } from '@blockchainApplication/manage/hooks';
-import { useGetNetworksMainChainStatus } from '@blockchainApplication/manage/hooks/queries/useGetNetworksMainChainStatus';
+import {
+  useApplicationManagement,
+  useCurrentApplication,
+} from '@blockchainApplication/manage/hooks';
+import { useNetworkStatus } from '@network/hooks/queries';
+import { useBlockchainApplicationMeta } from '@blockchainApplication/manage/hooks/queries/useBlockchainApplicationMeta';
+import { Client } from 'src/utils/api/client';
 import { PrimaryButton } from 'src/theme/buttons';
 
 const ApplicationBootstrap = ({ children }) => {
   const { mainChainNetwork } = useSettings('mainChainNetwork');
   const [isFirstTimeLoading, setIsFirstTimeLoading] = useState(true);
   const [, setCurrentApplication] = useCurrentApplication();
+  const { setApplications } = useApplicationManagement();
+  const queryClient = useRef(new Client({ http: mainChainNetwork?.serviceUrl }));
   useTransactionUpdate();
 
   const {
-    data: networksMainChainStatus,
-    isLoading: isVerifyingNetworkOptions,
-    isError: isErrorVerifyingNetworkOptions,
-  } = useGetNetworksMainChainStatus({ options: { enabled: !!mainChainNetwork } });
-
-  const selectedNetworkMainChain = networksMainChainStatus[mainChainNetwork?.name];
+    data: selectedNetworkStatus,
+    isLoading: isGettingNetworkStatus,
+    isError: isErrorGettingNetworkStatus,
+  } = useNetworkStatus({
+    options: { enabled: !!mainChainNetwork },
+    client: queryClient.current,
+  });
 
   const {
-    data: mainChainApplication,
+    data,
     isLoading: isGettingMainChain,
     isError: isErrorGettingMainChain,
     refetch: refetchMergedApplicationData,
-  } = useGetMergedApplication({
-    params: { chainID: selectedNetworkMainChain?.data?.chainID },
-    networkName: mainChainNetwork?.name,
-    isEnabled: !!selectedNetworkMainChain,
+  } = useBlockchainApplicationMeta({
```

Comment:
can we follow this pattern for hooks result.

```
const blockchainApplicationMetaResult = useBlockchainApplicationMeta({})
```

and then wherever we need to specific properties we can use `blockchainApplicationMetaResult. isLoading` or `blockchainApplicationMetaResult.data`, etc..

The above pattern will avoid the following:

- Incorrect naming of variables like `isLoading: isGettingMainChain` and
- The above approach is more readable and you can follow the code easily without knowing what `isGettingMainChain` belongs to